scripts/AgathasRoom.asm

```diff
@@ -104,29 +104,29 @@ AgathaScript3:
 	ld [wCurMapScript], a
 	ret
 
-AgathaScript2:
+AgathasRoomAgathaEndBattleScript:
 	call EndTrainerBattle
 	ld a, [wIsInBattle]
 	cp $ff
 	jp z, ResetAgathaScript
-	ld a, $1
+	ld a, TEXT_AGATHASROOM_AGATHA
 	ldh [hSpriteIndexOrTextID], a
 	call DisplayTextID
-	ld a, $1
+	ld a, SCRIPT_CHAMPIONSROOM_PLAYER_ENTERS
 	ld [wChampionsRoomCurScript], a
 	ret
 
-AgathasRoom_TextPointers:
-	dw AgathaText1
-	dw AgathaDontRunAwayText
+	def_text_pointers AgathasRoom
+	dw_const AgathasRoomAgathaScript,          TEXT_AGATHASROOM_AGATHA
+	dw_const AgathasRoomAgathaDontRunAwayText, TEXT_AGATHASROOM_AGATHA_DONT_RUN_AWAY
 
 AgathasRoomTrainerHeaders:
 	def_trainers
 AgathasRoomTrainerHeader0:
 	trainer EVENT_BEAT_AGATHAS_ROOM_TRAINER_0, 0, AgathaBeforeBattleText, AgathaEndBattleText, AgathaAfterBattleText
 	db -1 ; end
 
-AgathaText1:
+AgathasRoomAgathaScript:
```

Comment:
I argue this should be `AgathasRoomAgathaText`. It is a map text, not a map script. It is true that text can execute code, but this is always true for any text -- so it's not important to specify each time.

The distinction between the two tables of pointers is important. You can't `call` a map text; it has to be printed by the game's text processor. So, map texts should always have "Text" and not "Script" in their name.

Otherwise, a "text" that says it's a "script" is misleading about how it needs to be used, and so you'd have to constantly double check where each label comes from as you go.

This is what I meant in my last comment, sorry I should have been more clear.